### ISSUE TITLE
Example request: LEMP stack w/ custom image

### ISSUE THREAD

**Issue by mnpenner** [kubernetes/examples#380]
@wernight said it very well over [here](https://github.com/kubernetes/kubernetes/issues/17489#issue-117778148), so I'll copy-paste:

---

TL;DR Suggesting a LAMP example to help traditional users (those still running apt-get install on a single Linux server or knowing even less) migrate to Kubernetes.

After some experiments I got a broader idea of how things work. The current documentation isn't too bad but it's lacking of some common end to end case. For example to bring a standard LEMP (Linux + Nginx + PHP/Python/... + MySQL/Redis/...) to Kubernetes users need to read:

Use a Nginx image, use a MySQL image linked to a MySQL storage, and probably build a Docker image (by writing a Dockerfile) based on PHP.
Create a Container Registry and push the built image(s)
Deploy those images
Scale
I'd suggest a LAMP/LEMP tutorial example covering/answering points above in the follow way:

The docker-specific knowledge should be briefly covered with some pointers. The existing tutorials assume everything can be done using (existing) pre-built (public) images. This is often not the case and removes all the power of Docker and what comes with it.
Set up a Container Registry and push the built image(s)
This step has some elements but it's missing an example of Nginx + PHP/Python/... most examples I found are Apache with PHP in a single container which isn't the Docker way. I could help but I'd need some advice on choosing between:
Nginx + PHP-FPM (or similar) on the same Pod (i.e. multiple container Pod)
Nginx linking to PHP each on different Pods (so possibly one machine connecting to another)
Scaling of most LEMP solutions have the same common points, like having sessions stored in a distributed database. PHP/Django/... all allow that and require that to allow scaling. I'd just chose one of them for that example to have a practical example.
It could point of course to the existing Nginx example.

---

In particular, I don't want to use a pre-built image (I don't even understand why tutorials are suggesting it -- it's not a viable strategy is it? Even if you put your code in a "PVC", doesn't that defeat the purpose and kill the replicability?)

Do I *have* to push my image to a repository? Why? I've already created a cluster, can't I push it there? Should my images be versioned? Does that mean I have to keep updating my kubernetes yamls?
